kie-soup-dataset/kie-soup-dataset-sql/src/test/java/org/dashbuilder/dataprovider/sql/SQLDataSetProviderTest.java

```diff
@@ -0,0 +1,127 @@
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import org.dashbuilder.dataprovider.sql.dialect.Dialect;
+import org.dashbuilder.dataprovider.sql.model.Column;
+import org.dashbuilder.dataset.ColumnType;
+import org.dashbuilder.dataset.DataSetMetadata;
+import org.dashbuilder.dataset.def.DataSetDefFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.modules.junit4.PowerMockRunnerDelegate;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockRunnerDelegate(Parameterized.class)
+@PrepareForTest({JDBCUtils.class})
+public class SQLDataSetProviderTest {
+
+    @Parameters(name = "Dialect : {0}")
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] {
+                 {JDBCUtils.DB2, "FETCH FIRST 0 ROWS ONLY"}, 
+                 {JDBCUtils.DEFAULT, "LIMIT 0"},
+                 {JDBCUtils.H2, "LIMIT 0"},
+                 {JDBCUtils.MYSQL, "LIMIT 0"},
+                 {JDBCUtils.ORACLE, "FETCH FIRST 0 ROWS ONLY"},
+                 {JDBCUtils.POSTGRES, "LIMIT 0"},
+                 {JDBCUtils.SQLSERVER, "TOP 0"},
+                 {JDBCUtils.SYBASE_ASE, "TOP 0"}
+           });
+    }
+    
+    @Mock
+    Statement statement;
+    
+    @Mock
+    ResultSet resultSet;
+    
+    @Spy
+    SQLDataSetProvider dataSetProvider = SQLDataSetProvider.get();
+    
+    ResultSetHandler resultSetHandler;
+    
+    private final Dialect dialect;
+    private final String limitAssertion;
+    
+    public SQLDataSetProviderTest(Dialect dialect, String limitAssertion) {
+        this.dialect = dialect;
+        this.limitAssertion = limitAssertion;
+    }
+    
+    @Before
+    public void setUp() throws Exception {
+        PowerMockito.mockStatic(JDBCUtils.class);
+        
+        final List<Column> dbColumns = new ArrayList<Column>();
+        dbColumns.add(SQLFactory.column("dbWins1", ColumnType.LABEL, 10));
+        dbColumns.add(SQLFactory.column("dbWins2", ColumnType.LABEL, 10));
+        dbColumns.add(SQLFactory.column("dbWins3", ColumnType.LABEL, 10));
+
+        dataSetProvider.setDataSourceLocator(new DatabaseTestSettings().getDataSourceLocator());
+        resultSetHandler = new ResultSetHandler(resultSet, statement);
+        
+        when(JDBCUtils.dialect(any(Connection.class))).thenReturn(dialect);
+        when(JDBCUtils.executeQuery(any(Connection.class), any())).thenReturn(resultSetHandler);
+        when(JDBCUtils.getColumns(any(),any())).thenReturn(dbColumns);
+        when(JDBCUtils.fixCase(any(Connection.class),eq("test"))).thenReturn("TEST");
```

Comment:
This is not needed as we in the end use just one Connection with different dialects, so the outcome is still the same.

Reply:
I'd leave it to have a more meaningful sql (with table name):
_SELECT * FROM TEST LIMIT 0_ vs. _SELECT * FROM null LIMIT 0_

Reply:
ah, you are right, I forgot that this is mocked as well :)